satellite/orders/endpoint.go

```diff
@@ -161,6 +181,30 @@ func (endpoint *Endpoint) Settlement(stream pb.Orders_SettlementServer) (err err
 			}
 		}
 
+		if err := endpoint.DB.UpdateBucketBandwidthSettle(ctx, bucketID, orderLimit.Action, order.Amount); err != nil {
+			err := stream.Send(&pb.SettlementResponse{
+				SerialNumber: orderLimit.SerialNumber,
+				// TODO should we reject or accept such error
+				Status: pb.SettlementResponse_REJECTED,
+			})
+			if err != nil {
+				return formatError(err)
+			}
+			return nil
+		}
+
+		if err := endpoint.DB.UpdateStoragenodeBandwidthSettle(ctx, orderLimit.StorageNodeId, orderLimit.Action, order.Amount); err != nil {
+			err := stream.Send(&pb.SettlementResponse{
```

Comment:
When we hit error here, we should "unuse" the serial number and then return the error and not reply with neither accepted nor rejected. That will cause the storage node to retry later.

Reply:
It makes sense for first `UpdateBucketBandwidthSettle`. If we will unuse serial number here then with another try bucket bandwidth will be updated second time.

Reply:
for the second one yeah, we would also need to rollback the previous one

Reply:
fixed